prover/prover_dal/src/fri_witness_generator_dal.rs

```diff
@@ -837,6 +847,72 @@ impl FriWitnessGeneratorDal<'_, '_> {
         .collect()
     }
 
+    pub async fn move_recursion_tip_jobs_from_waiting_to_queued(&mut self) -> Vec<u64> {
+        sqlx::query!(
+            r#"
+            UPDATE recursion_tip_witness_jobs_fri
+            SET
+                status = 'queued'
+            WHERE
+                l1_batch_number IN (
+                    SELECT
+                        prover_jobs_fri.l1_batch_number
+                    FROM
+                        prover_jobs_fri
+                        JOIN recursion_tip_witness_jobs_fri rtwj ON prover_jobs_fri.l1_batch_number = rtwj.l1_batch_number
+                    WHERE
+                        rtwj.status = 'waiting_for_proofs'
+                        AND prover_jobs_fri.status = 'successful'
+                        AND prover_jobs_fri.aggregation_round = 2
+                        AND prover_jobs_fri.is_node_final_proof = true
+                    GROUP BY
+                        prover_jobs_fri.l1_batch_number,
+                        rtwj.number_of_final_node_jobs
+                    HAVING
+                        COUNT(*) = rtwj.number_of_final_node_jobs
+                )
+            RETURNING
+                l1_batch_number;
+            "#,
+    )
+    .fetch_all(self.storage.conn())
+    .await
+    .unwrap()
+    .into_iter()
+    .map(|row| (row.l1_batch_number as u64))
+    .collect()
+    }
+
+    pub async fn move_scheduler_jobs_from_waiting_to_queued(&mut self) -> Vec<u64> {
+        sqlx::query!(
+            r#"
+            UPDATE scheduler_witness_jobs_fri
+            SET
+                status = 'queued'
+            WHERE
+                l1_batch_number IN (
+                    SELECT
+                        prover_jobs_fri.l1_batch_number
+                    FROM
+                        prover_jobs_fri
+                        JOIN scheduler_witness_jobs_fri swj ON prover_jobs_fri.l1_batch_number = swj.l1_batch_number
+                    WHERE
+                        swj.status = 'waiting_for_proofs'
+                        AND prover_jobs_fri.status = 'successful'
+                        AND prover_jobs_fri.aggregation_round = 3
+                )
+            RETURNING
+                l1_batch_number;
+            "#,
+    )
+    .fetch_all(self.storage.conn())
+    .await
+    .unwrap()
+    .into_iter()
+    .map(|row| (row.l1_batch_number as u64))
+    .collect()
```

Comment:
Nit: Incorrect indentation here (should be 4 spaces greater).

Reply:
Rust formatter is bad with handling code inside macros and code surrounding macros (like in this case). Which is natural because in the general case, macros can have arbitrary syntax.